src/Interface.jl

```diff
@@ -44,19 +47,49 @@ function marginal(x::ManifoldKernelDensity{M,B},
                   dims::AbstractVector{<:Integer}  ) where {M <: AbstractManifold , B}
   #
   ldims::Vector{Int} = collect(dims)
-  ManifoldKernelDensity(x.manifold, x.belief, ldims)
+  ManifoldKernelDensity(x.manifold, x.belief, ldims, x._u0)
 end
 
 function marginal(x::ManifoldKernelDensity{M,B,L}, 
                   dims::AbstractVector{<:Integer}  ) where {M <: AbstractManifold , B, L <: AbstractVector{<:Integer}}
   #
   ldims::Vector{Int} = collect(L[dims])
-  ManifoldKernelDensity(x.manifold, x.belief, ldims)
+  ManifoldKernelDensity(x.manifold, x.belief, ldims, x._u0)
 end
 # manis = convert(Tuple, x.manifold)
 # partMani = _reducePartialManifoldElements(manis[dims])
 # pts = getPoints(x)
 
+
+"""
+    $SIGNATURES
+
+Helper function to convert coordinates to a desired on-manifold point.
+
+Notes
+- `u0` is used to identify the data type for a point
+- Pass in a different `exp` if needed.
+"""
+makeManifoldPoint(M::MB.AbstractManifold,
+                  coords::AbstractVector{<:Real},
+                  u0=zeros(0),
+                  ϵ=identity(M,u0),
+                  _exp::Function=exp ) = _exp(M, ϵ, hat(M, ϵ, coords))
```

Comment:
Just as reference, it look like a more general way to do this is to use:
`retraction_method::AbstractRetractionMethod=ExponentialRetraction()`
`retract(M, p, X, retraction_method)`

Reply:
retract it is!